gammapy/spectrum/energy.py

```diff
@@ -0,0 +1,293 @@
+        if unit is None:
+            raise ValueError("No energy unit could be found in the header of."
+                             "{0}. Please specifiy a unit".format(header.name))
+
+        energy = Quantity(hdu.data['Energy'], unit)
+        return Energy(energy)
+
+    def to_fits(self, **kwargs):
+        """Write ENERGIES fits extension
+
+        Returns
+        -------
+        hdu: `~astropy.io.fits.BinTableHDU`
+            ENERGIES fits extension
+
+
+        """
+
+        col1 = fits.Column(name='Energy', format='D', array=self.value)
+        cols = fits.ColDefs([col1])
+        hdu = fits.BinTableHDU.from_columns(cols)
+        hdu.name = 'ENERGIES'
+        hdu.header['TUNIT1'] = "{0}".format(self.unit)
+
+        return hdu
+
+
+class EnergyBounds(Energy):
+
+    """Energy bin edges
+
+    Stored as "EBOUNDS" FITS table extensions.
+
+    Parameters
+    ----------
+
+    energy : `~numpy.array`, scalar, `~astropy.units.Quantity`
+        EnergyBounds
+    unit : `~astropy.units.UnitBase`, str
+        The unit of the values specified for the energy.  This may be any
+        string that `~astropy.units.Unit` understands, but it is better to
+        give an actual unit object.
+
+    """
+
+    def __new__(cls, energy, unit=None, dtype=None, copy=True):
+
+        self = super(EnergyBounds, cls).__new__(cls, energy, unit,
+                                                dtype=dtype, copy=copy)
+
+        self._nbins = self.size - 1
+
+        return self
+
+    def __array_finalize__(self, obj):
+        if obj is None:
+            return
+        self._nbins = self.size - 1
+        self._unit = getattr(obj, '_unit', None)
+
+    @property
+    def log_centers(self):
+        """Log centers of the energy bounds
+        """
+        
+        center = np.sqrt(self[:-1] * self[1:])
+        return Energy(center)
+
+    @property
+    def upper_bounds(self):
+        """Upper energy bin edges
+        """
+        upper = self[1:]
+        return EnergyBounds(upper)
+
+    @property
+    def lower_bounds(self):
+        """Lower energy bin edges
+        """
+        
+        lower = self[:-1]
+        return EnergyBounds(lower)
+
+    @staticmethod
+    def equal_log_spacing(emin, emax, nbins, unit=None):
+        """Create EnergyBounds with equal log-spacing.
+
+        If no unit is given, it will be taken from emax
+
+        Parameters
+        ----------
+        emin : `~astropy.units.Quantity`, float
+            Lowest energy bin
+        emax : `~astropy.units.Quantity`, float
+            Highest energy bin
+        bins : int
+            Number of bins
+        unit : `~astropy.units.UnitBase`, str
+            Energy unit
+
+        Returns
+        -------
+        EnergyBounds
+        """
+
+        bounds = super(EnergyBounds, EnergyBounds).equal_log_spacing(
+            emin, emax, nbins + 1, unit)
+
+        return bounds.view(EnergyBounds)
+
+    @staticmethod
+    def from_fits(hdu, unit=None):
```

Comment:
Same comments as for `Energy.from_fits`; think that as a class method, you would not have to redefine it at all.